### PR TITLE
Add Cubist

### PR DESCRIPTION
**Pull request by asmaahassan90**

### CODE REVIEW

File: var/spack/repos/builtin/packages/cubist/package.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    """Cubist is a powerful tool for generating rule-based models that
+    balance the need for accurate prediction against the requirements of
+    intelligibility. Cubist models generally give better results than those
+    produced by simple techniques such as multivariate linear regression,
+    while also being easier to understand than neural networks."""
+
+    homepage = "https://www.rulequest.com"
+    url      = "https://www.rulequest.com/GPL/Cubist.tgz"
+
+    version('2.07', 'f2b20807cd3275e775c42263a4efd3f50df6e495a8b6dc8989ea2d41b973ac1a')
```
Comment (NasrHassanein) [inline var/spack/repos/builtin/packages/cubist/package.py:22]: The URL provided will always download the latest version, This hash will fail if the app gets updated.

Comment (asmaahassan90) [reply var/spack/repos/builtin/packages/cubist/package.py:22]: It's the only link provided though their download page: https://www.rulequest.com/download.html

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/cubist/package.py:22]: Unfortunately, we have many packages like this. For example, the `triangle` package has the same problem. If you look at https://www.rulequest.com/GPL/, Cubist hasn't been updated in almost a decade, so I don't think we'll have to worry too much about frequent hash changes.

File: var/spack/repos/builtin/packages/cubist/package.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack import *
+
+
+class Cubist(Package):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cubist/package.py]: Can you convert this to a `MakefilePackage`? Then `patch(self)` will become `edit(self, spec, prefix)` and `make()` can be removed from the `install` method.

Comment (asmaahassan90) [reply var/spack/repos/builtin/packages/cubist/package.py]: done

**Pull request merged**